src/functionalTest/java/uk/gov/hmcts/reform/locationrefdata/cameltest/LrdBatchApplicationTest.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfiguration;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlConfig;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+import uk.gov.hmcts.reform.data.ingestion.configuration.AzureBlobConfig;
+import uk.gov.hmcts.reform.data.ingestion.configuration.StorageCredentials;
+import uk.gov.hmcts.reform.locationrefdata.camel.binder.ServiceToCcdCaseType;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.LrdBatchIntegrationSupport;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.RestartingSpringJUnit4ClassRunner;
+import uk.gov.hmcts.reform.locationrefdata.cameltest.testsupport.SpringRestarter;
+import uk.gov.hmcts.reform.locationrefdata.config.LrdCamelConfig;
+import uk.gov.hmcts.reform.locationrefdata.configuration.BatchConfig;
+
+import java.io.FileInputStream;
+
+import static org.springframework.util.ResourceUtils.getFile;
+
+@TestPropertySource(properties = {"spring.config.location=classpath:application-integration.yml,"
+    + "classpath:application-leaf-integration.yml"})
+@RunWith(RestartingSpringJUnit4ClassRunner.class)
+@MockEndpoints("log:*")
+@ContextConfiguration(classes = {LrdCamelConfig.class, CamelTestContextBootstrapper.class,
+    JobLauncherTestUtils.class, BatchConfig.class, AzureBlobConfig.class, StorageCredentials.class},
+    initializers = ConfigFileApplicationContextInitializer.class)
+@SpringBootTest
+@EnableAutoConfiguration(exclude = JpaRepositoriesAutoConfiguration.class)
+@EnableTransactionManagement
+@SqlConfig(dataSource = "dataSource", transactionManager = "txManager",
+    transactionMode = SqlConfig.TransactionMode.ISOLATED)
+@SuppressWarnings("unchecked")
+public class LrdBatchApplicationTest extends LrdBatchIntegrationSupport {
+
+    @Value("${start-route}")
+    private String startRoute;
+
+    @Value("${archival-route}")
+    String archivalRoute;
+
+    @Before
+    public void init() {
+        jdbcTemplate.execute(truncateAudit);
+        SpringRestarter.getInstance().restart();
+    }
+
+    @Test
+    @Sql(scripts = {"/testData/truncate-lrd.sql"})
+    public void testTaskletSuccess() throws Exception {
+        testInsertion();
+    }
+
+    @Test
+    @Sql(scripts = {"/testData/truncate-lrd.sql"})
+    public void testTaskletSuccessWithUpdateAndDelete() throws Exception {
+
+        testInsertion();
+        jdbcTemplate.execute("delete from DATALOAD_SCHEDULAR_AUDIT");
```

Comment:
i dont think we should execute this delete statement here

Reply:
This is inser and update and we have idempotent job how you test day2 scenario? this is needed.

Reply:
You can do it in multiple ways.. You are only asserting by checking record in  audit table which you can do by doing a select with order by statement as well.

Reply:
What is wrong in this way?? I don't get that. delete the record in test won't harm